mitodl/open-discussions
# Fixed id assignment during indexing

#### What are the relevant tickets?
Fixes #780

#### What's this PR do?
Assigns explicit ids for posts and comments during indexing. Check the issue for the reasoning.

#### How should this be manually tested?
- Run `docker-compose run web ./manage.py recreate_index`
- Check the documents in the index and make sure the ids look right for posts (`p_[reddit_id]`) and comments (`c_[reddit_id]`)

`curl` can be used from the CLI to easily inspect documents: 

```
curl -XGET -H "Content-Type: application/json" -s 'http://localhost:9101/discussions_local_default/_search?pretty=true'
```

#### Where should the reviewer start?
`search/serializers.py`

#### Any background context you want to provide?
⚠️ `search/api.py` and `search/api_test.py` (and related changed) were added as part of #739, which is under review but not merged yet ⚠️
